### ISSUE TITLE
RuntimeException

### ISSUE THREAD

**Issue by lizardfreak321** [InventivetalentDev/CompactNPCLib#23]
## Explanation

Based on the error log, it appears that the setYaw method is causing an error. It happens on 1.8.8 but not 1.10.2
## What version of the plugin are you using? _Type /version &lt;Plugin Name&gt;_

2.2.6
## What Spigot version are you using? _Type /version_

CraftBukkit version git-Spigot-e4d4710-e1ebe52 (MC: 1.8.8) (Implementing API version 1.8.8-R0.1-SNAPSHOT)
## Do you have an error log? Use [pastebin.com](http://pastebin.com). _If you're not sure, upload your whole server log_

http://pastebin.com/CvFQQxhq
## Did your client crash? _Upload errors in .minecraft/logs/latest.log as well_

It did not, but it did disconnect me.
## Additional information?  _(Are you using Bungeecord? Did it work in previous versions? etc.)_

I am using bungeecord. It works on 1.10.2 where I have originally been testing it.

**Comment by lizardfreak321** [comment]
Excuse me, @InventivetalentDev, but it has been 19 days and this project is at a standstill because of this issue.
It appears that the cause is that the method doesn't exist in the NMS code before 1.9. I have had trouble finding a workaround by calling NMS methods myself, but I am inexperienced with NMS code.

**Comment by lizardfreak321** [comment]
@InventivetalentDev I have waited several more weeks for this to get fixed, it has been over 2 months since I reported it. I need it fixed for my plugin to work, and getting this plugin to build is surprisingly difficult.
I know you have lost interest in developing, but can you please attempt to fix this feature and possibly accept the pull request?

**Comment by lizardfreak321** [comment]
All you should need to do is copy the code you had in the old NPCLib into CompactNPCLib. Can you please do this?
`@Override
	public void setYaw(float yaw) {
		try {
			NMSClass.nmsEntity.getDeclaredField("yaw").set(this.theEntity, NMSUtil.clampYaw(yaw));

			NMSClass.nmsEntityLiving.getDeclaredField(Minecraft.VERSION.olderThan(Minecraft.Version.v1_8_R1) ? "aP" : NPCLib.getServerVersion() <= 181 ? "aJ" : NPCLib.getServerVersion() <= 183 ? "aL" : "aP").set(this.theEntity, yaw);
			NMSClass.nmsEntityLiving.getDeclaredField(Minecraft.VERSION.olderThan(Minecraft.Version.v1_8_R1) ? "aL" : NPCLib.getServerVersion() <= 181 ? "aF" : NPCLib.getServerVersion() <= 182 ? "aH" : NPCLib.getServerVersion() <= 183 ? "aG" : "aL").set(this.theEntity, yaw);
			NMSClass.nmsEntityLiving.getDeclaredField(Minecraft.VERSION.olderThan(Minecraft.Version.v1_8_R1) ? "aK" : NPCLib.getServerVersion() <= 181 ? "aE" : NPCLib.getServerVersion() <= 183 ? "aG" : "aK").set(this.theEntity, yaw);

			/* This seems to be the 'real' field */
			NMSClass.nmsEntityLiving.getDeclaredField(Minecraft.VERSION.olderThan(Minecraft.Version.v1_8_R1) ? "aO" : NPCLib.getServerVersion() <= 181 ? "aI" : NPCLib.getServerVersion() <= 183 ? "aK" : "aO").set(this.theEntity, yaw);
		} catch (Exception e) {
			e.printStackTrace();
		}
	}
`
You just need to add that to lines 77-84 in NPCLivingAbstract.java in CompactNPCLib.

If you don't want to do that please just provide how to get this to build in Eclipse and I will try to submit a pull request?
I'm not asking you to overhaul a bunch of code to get it to work on a new version, all I am asking is for is you to fix a little thing that is preventing my plugin from working.

**Comment by InventivetalentDev** [comment]
You'll have to use IntelliJ to import/compile the project. 
The code from the old NPCLib was literally just a guessing game on which fields do what, and I still have no clue if anything there actually did what I hoped it would. Which is why I updated it to a simpler version here.  

Apparently Spigot 1.8.8 doesn't have that field, I'll leave it up to you to figure out which fields are responsible there (and also in 1.11), since I feel like I've wasted enough time on that, sorry.
